app/forms/CommonBinders.scala

```diff
@@ -16,17 +16,61 @@
 
 package forms
 
+import java.time.LocalDate
+
 import play.api.data.{FormError, Forms, Mapping}
 import play.api.data.format.Formatter
+import utils.Constants
+import common.Dates._
+import common.Exceptions
+import common.Validation._
 
 import scala.util.{Failure, Success, Try}
 
 trait CommonBinders {
 
   private val EMPTY_STRING = ""
+  private val DAY_LIMIT = 31
+  private val MONTH_LIMIT = 12
+
+  implicit val bigDecimalFormatter = new Formatter[Option[BigDecimal]] {
+    def bind(key: String, data: Map[String, String]): Either[Seq[FormError], Option[BigDecimal]] =
+      if (!checkPreviousKey(key.substring(0, key.length - 3), data).getOrElse(true)){
+        data.get(key).map(validateNumber(key, _)).getOrElse(Left(Seq(FormError(key, "pla.base.errors.errorQuestion", Nil))))
+      } else Right(None)
+
+    def unbind(key: String, value: Option[BigDecimal]): Map[String, String] = {
+      value match {
+        case Some(data) => Map(key -> data.toString())
+        case None => Map()
+      }
+    }
+  }
+  val requiredBigDecimal: Mapping[Option[BigDecimal]] = Forms.of[Option[BigDecimal]](bigDecimalFormatter)
```

Comment:
Maybe rename this to yesNoBigDecimal or conditionallyRequiredBigDecimal or something.